supporting-blog-content/lexical-and-semantic-search-with-elasticsearch/ecommerce_dense_sparse_project.ipynb

```diff
@@ -0,0 +1,834 @@
+        "            },\n",
+        "            mappings={\n",
+        "# Saving disk space by excluding the ELSER tokens and the dense_vector field from document source.\n",
+        "# Note: That should only be applied if you are certain that reindexing will not be required in the future.\n",
+        "            \"_source\" : {\n",
+        "            \"excludes\": [\"ml.tokens\",\"description_vector.predicted_value\"]\n",
+        "           },\n",
+        "        \"properties\": {\n",
+        "        \"product\": {\n",
+        "        \"type\": \"text\",\n",
+        "        \"fields\": {\n",
+        "          \"keyword\": {\n",
+        "            \"type\": \"keyword\",\n",
+        "            \"ignore_above\": 256\n",
+        "          }\n",
+        "        }\n",
+        "      },\n",
+        "      \"description\": {\n",
+        "        \"type\": \"text\",\n",
+        "        \"fields\": {\n",
+        "          \"keyword\": {\n",
+        "            \"type\": \"keyword\",\n",
+        "            \"ignore_above\": 256\n",
+        "          }\n",
+        "        }\n",
+        "      },\n",
+        "      \"category\": {\n",
+        "        \"type\": \"text\",\n",
+        "        \"fields\": {\n",
+        "          \"keyword\": {\n",
+        "            \"type\": \"keyword\",\n",
+        "            \"ignore_above\": 256\n",
+        "          }\n",
+        "        }\n",
+        "      },\n",
+        "      \"ml.tokens\": { # The name of the field to contain the generated tokens.\n",
+        "        \"type\": \"rank_features\" # ELSER output must be ingested into a field with the rank_features field type.\n",
+        "      },\n",
+        "     \"description_vector.predicted_value\": { # Inference results field, target_field.predicted_value\n",
+        "     \"type\": \"dense_vector\",\n",
+        "     \"dims\": 768, # The all-mpnet-base-v2 model has embedding_size of 768, so dims is set to 768.\n",
+        "     \"index\": \"true\",\n",
+        "     \"similarity\": \"dot_product\" #  When indexing vectors for approximate kNN search, you need to specify the similarity function for comparing the vectors.\n",
+        " }\n",
+        "  }\n",
+        "\n",
+        "}\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Load documents\n",
+        "\n",
+        "Then we load `products-ecommerce.json` into the `ecommerce` index."
+      ],
+      "metadata": {
+        "id": "Vo-LKu8TOT5j"
+      },
+      "id": "Vo-LKu8TOT5j"
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "3cfdc3b7-7e4f-4111-997b-c333ac8938ba",
+      "metadata": {
+        "id": "3cfdc3b7-7e4f-4111-997b-c333ac8938ba"
+      },
+      "outputs": [],
+      "source": [
+        "#  dataset\n",
+        "\n",
+        "url = \".../products-ecommerce.json\" # json raw file - update the link here\n",
```

Comment:
will need the full http url unfortunately. 
ValueError: unknown url type: '.../products-ecommerce.json'

Ran it on google colab [here](https://colab.research.google.com/github/elastic/elasticsearch-labs/blob/priscillaparodi-patch-1/supporting-blog-content/lexical-and-semantic-search-with-elasticsearch/ecommerce_dense_sparse_project.ipynb)

Reply:
added the json file here, updated the url.